jeneamocan/script.rb

```diff
@@ -1,204 +1,184 @@
-    store
+  def set_date
+    day = Date.today.prev_month(2).day.to_s
+    browser.input(name: 'from').click
+    browser.a(class: %w"ui-datepicker-prev ui-corner-all").click
+    browser.a(text: "#{day}").click
   end
 
-  def last_transaction
-    authentication_check
-    puts "Fetching latest transaction"
-    browser.goto("https://web.vb24.md/wb/#menu/MAIN_215.CP_HISTORY")
-    month = browser.div(class: "month-delimiter").text
-    day = browser.div(class: "day-header").text.split[0]
-    time = browser.span(class: "history-item-time").text
-    date = day + " " + month + " " + time
-    description = browser.a(class: "operation-details").text
-    amount = browser.span(class: %w"history-item-amount transaction").text
-    {"transactions" => [{date: date, description: description, amount: amount}]}
+  def add_transactions
+    @accounts.each do |acount|
+      @transactions.each do |transaction|
+        acount.transactions << transaction
+      end
+    end
   end
 
-  def transactions_2months
-    authentication_check
-    browser.goto("https://web.vb24.md/wb/#menu/MAIN_215.CP_HISTORY")
-    set_date
-    puts "Fetching transactions for the last two months"
-    sleep(2)
-    page = parse
-    hash = {"transactions" => []}
-    page.css('li[class="history-item success "]').map do |i|
-      month = i.xpath('../../preceding-sibling::div[@class = "month-delimiter"]').last.text
-      day = i.parent.parent.css('div[class = "day-header"]').text
-      time = i.css('span[class= "history-item-time"]').text
-      date = day + " " + month + " " + time
-      description = i.css('span[class="history-item-description"]').text.split.join(" ")
-      if not i.css('span[class="history-item-amount transaction income"]').text.empty?
-        amount = i.css('span[class="history-item-amount transaction income"]').text
-      elsif not i.css('span[class="history-item-amount total"]').text.empty?
-        amount = i.css('span[class="history-item-amount total"]').text
-      elsif not i.css('span[class="history-item-amount transaction "]').text.empty?
-        amount = i.css('span[class="history-item-amount transaction "]').text
-      else 
+  def assemble
+    hash = Hash.new
+    @accounts.map do |account|
+      account_hash = Hash.new
+      account_hash['name'] = account.name
+      account_hash['balance'] = account.name
+      account_hash['currency'] = account.currency
+      account_hash['nature'] = account.nature
+      account_hash['transactions'] = []
+      account.transactions.map do |transaction|
+        transaction_hash = Hash.new
+        transaction_hash['date'] = transaction.date
+        transaction_hash['description'] = transaction.description
+        transaction_hash['amount'] = transaction.amount
+        account_hash['transactions'].push transaction_hash
       end
-    res = {date: date, description: description, amount: amount}
-    hash["transactions"] << res
-    end
+    hash["accounts"] = account_hash
     hash
```

Comment:
Something wrong with the indentation. There is no possible way you there can be 2 lines of code one of which is `end` and they are still lined up.